ecommerce_cms/src/store/index.js

```diff
@@ -0,0 +1,124 @@
+import Vue from 'vue'
+import Vuex from 'vuex'
+import axios from 'axios'
+import router from '../router/index'
+
+Vue.use(Vuex)
+
+export default new Vuex.Store({
+  state: {
+    notification: '',
+    products: []
+  },
+  mutations: {
+    SET_NOTIFICATION (state, payload) {
+      state.notification = payload
+    },
+    SET_PRODUCT (state, payload) {
+      state.products = payload
+    }
+  },
+  actions: { // nge fetch products udah pasti di action
+    register (context, payload) {
+      const email = payload.email
+      const password = payload.password
+      return axios({
+        method: 'POST',
+        url: 'https://peaceful-fortress-31291.herokuapp.com/admin/register',
+        data: {
+          email,
+          password
+        }
+      })
+    },
+    login (context, payload) {
+      // context object, context.state, context.commit, context.dispatch
+      const email = payload.email
+      const password = payload.password
+      return axios({
+        method: 'POST',
+        url: 'https://peaceful-fortress-31291.herokuapp.com/admin/login',
+        data: {
+          email,
+          password
+        }
+      })
+    },
+    logout (context) {
```

Comment:
kalo syncronous kaya gini ga perlu bikin di action tapi dari komponen langsung aja akses this.$store.commit